src/main/java/BadPracticesModel.java

```diff
@@ -0,0 +1,68 @@
+import lombok.*;
+
+import java.util.List;
+
+/**
+ * @author madhankumar
+ */
+@Setter
+@Getter
+@ToString
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class BadPracticesModel {
+    private String model;
+    private List<?> messages;
+    private int n;
+    private double temperature;
+
+    // Unused variable
+    private String unusedVariable = "I am unused";
+
+    public boolean newMethod(String st1, String st2) {
+        // Using '==' for string comparison instead of 'equals'
+        if (st1 == st2) {
```

Comment:
Using '==' for string comparison is incorrect. It should use '.equals()' to compare string values. (Severity: high)